Tasks/AzureRmWebAppDeploymentV3/operations/KuduServiceUtility.ts

```diff
@@ -53,7 +53,7 @@ export class KuduServiceUtility {
             await this._appServiceKuduService.uploadFile(directoryPath, 'kuduPostDeploymentScript_' + uniqueID + fileExtension, scriptFile.filePath);
             console.log(tl.loc('ExecuteScriptOnKudu'));
             await this.runCommand(directoryPath,
-             'mainCmdFile_' + uniqueID + fileExtension + ' ' + uniqueID,
+             (taskParams.isLinuxApp ? 'sh ' : '') + 'mainCmdFile_' + uniqueID + fileExtension + ' ' + uniqueID,
```

Comment:
In Linux we have 2 options to execute script -
1. source file.sh  - It is used when you want to executed in the current shell context, so any changes made to your environment will be visible when the script finishes execution. Script doesn't need execute permission.
2. sh file.sh or ./file.sh - Execute the script in a subshell (Create sub-process). The variables, functions and aliases created in this subshell are only known to the particular bash session of that subshell. When that shell exits and the parent regains control, everything is cleaned up and all changes to the state of the shell made by the script, are forgotten.

Not sure about what is the requirement on kudu side ?

Reply:
@RoshanKumarMicrosoft , the script is meant to use the variables already defined in the environment and not appending new ones, so we chose the second option. 
Also, kudu does the same thing. It doesnt save any state / variables that are defined in the post deployment script.